app/controllers/concerns/contextual_comms_ab_testable.rb

```diff
@@ -0,0 +1,182 @@
+      allowed_variants: %w(NoCampaign BlueBoxCampaign NativeCampaign),
+      control_variant: 'NoCampaign'
+    )
+  end
+
+  def contextual_comms_test_variant
+    @contextual_comms_test_variant ||= contextual_comms_test.requested_variant(request.headers)
+  end
+
+  def set_test_response_header
+    contextual_comms_test_variant.configure_response(response)
+  end
+
+  def show_contextual_comms_campaign?
+    !contextual_comms_test_variant.variant?("NoCampaign") && campaign_name.present?
+  end
+
+  def show_blue_box_campaign?
+    contextual_comms_test_variant.variant?("BlueBoxCampaign")
+  end
+
+  def show_native_campaign?
+    contextual_comms_test_variant.variant?("NativeCampaign")
+  end
+
+  def campaign_name
+    @campaign_name ||=
+      if get_in_go_far_pages.include?(content_item_path)
+        :get_in_go_far
+      elsif eating_pages.include?(content_item_path)
+        :eating
+      elsif check_your_pay_pages.include?(content_item_path)
+        :check_your_pay
+      elsif your_pension_pages.include?(content_item_path)
+        :your_pension
+      end
+  end
+
+  def campaign_link
+    CAMPAIGN_DATA[campaign_name][:link]
+  end
+
+  def campaign_title
+    CAMPAIGN_DATA[campaign_name][:title]
+  end
+
+  def campaign_description
+    CAMPAIGN_DATA[campaign_name][:description]
+  end
+
+  def get_in_go_far_pages
+    %w(
+      /career-skills-and-training
+      /mature-student-university-funding
+      /higher-education-courses-find-and-apply
+      /what-different-qualification-levels-mean
+      /what-different-qualification-levels-mean/list-of-qualification-levels
+      /what-different-qualification-levels-mean/compare-different-qualification-levels
+      /further-education-courses
+      /improve-english-maths-it-skills
+      /further-education-courses
+      /further-education-courses/financial-help
+      /further-education-courses/find-a-course
+      /looking-for-work-if-disabled
+      /exoffenders-and-employment
+    )
+  end
+
+  def eating_pages
+    %w(
+      /child-trust-funds
+      /child-trust-funds/accounts-for-looked-after-children
+      /child-trust-funds/add-money-to-the-account
+      /child-trust-funds/managing-the-account
+      /free-school-transport
+      /healthy-start
+      /healthy-start/eligibility
+      /healthy-start/how-to-claim
+      /healthy-start/what-youll-get
+      /help-with-childcare-costs
+      /help-with-childcare-costs/childcare-vouchers
+      /help-with-childcare-costs/free-childcare-2-year-olds
+      /help-with-childcare-costs/free-childcare-2-year-olds-benefits
+      /help-with-childcare-costs/free-childcare-and-education-for-2-to-4-year-olds
+      /help-with-childcare-costs/support-while-you-study
+      /help-with-childcare-costs/tax-credits
+      /help-with-childcare-costs/tax-free-childcare
+      /help-with-childcare-costs/universal-credit
+      /school-uniform
+    )
+  end
+
+  def check_your_pay_pages
+    %w(
```

Comment:
It would be better in terms of performance to freeze these as constants and (optionally) return them via the methods.